File: services/inventory-service/src/main/java/io/minishop/inventory/web/dto/ReservationResponse.java
package io.minishop.inventory.web.dto;

import io.minishop.inventory.domain.InventoryReservation;
import io.minishop.inventory.domain.ReservationStatus;

import java.time.Instant;

public record ReservationResponse(
        Long id,
        Long productId,
        Long orderId,
        Integer quantity,
        ReservationStatus status,
        Instant createdAt,
        Instant releasedAt,
        boolean idempotent
) {
    public static ReservationResponse from(InventoryReservation r, boolean idempotent) {
        return new ReservationResponse(
                r.getId(), r.getProductId(), r.getOrderId(), r.getQuantity(),
                r.getStatus(), r.getCreatedAt(), r.getReleasedAt(), idempotent
        );
    }
}

File: services/inventory-service/src/main/kotlin/io/minishop/inventory/web/dto/ReservationResponse.kt
package io.minishop.inventory.web.dto

import io.minishop.inventory.domain.InventoryReservation
import io.minishop.inventory.domain.ReservationStatus
import java.time.Instant

@JvmRecord
data class ReservationResponse(
    val id: Long?,
    val productId: Long?,
    val orderId: Long?,
    val quantity: Int?,
    val status: ReservationStatus?,
    val createdAt: Instant?,
    val releasedAt: Instant?,
    val idempotent: Boolean,
) {
    companion object {
        @JvmStatic
        fun from(r: InventoryReservation, idempotent: Boolean): ReservationResponse =
            ReservationResponse(
                r.id, r.productId, r.orderId, r.quantity,
                r.status, r.createdAt, r.releasedAt, idempotent,
            )
    }
}
